crates/biome_markdown_parser/src/lexer/mod.rs

```diff
@@ -263,6 +269,39 @@ impl<'src> MarkdownLexer<'src> {
         TAB
     }
 
+    fn consume_thematic_break_literal(&mut self) -> MarkdownSyntaxKind {
+        self.assert_at_char_boundary();
+
+        let start_char = match self.current_byte() {
+            Some(b'-') => b'-',
+            Some(b'*') => b'*',
+            Some(b'_') => b'_',
+            _ => return self.consume_textual(),
+        };
+
+        let mut count = 0;
+        loop {
+            self.skip_whitespace();
+            if matches!(self.current_byte(), Some(ch) if ch == start_char) {
+                self.advance(1);
+                count += 1;
+            } else {
+                break;
+            }
+        }
+        // until next newline or eof
+        if matches!(self.current_byte(), Some(b'\n' | b'\r') | None) && count >= 3 {
+            return MD_THEMATIC_BREAK_LITERAL;
+        }
+        ERROR_TOKEN
+    }
+
+    fn skip_whitespace(&mut self) {
+        while matches!(self.current_byte(), Some(b'\t' | b' ')) {
+            self.advance(1);
+        }
```

Comment:
The lexer has already a function to consume whitespace, let's use that please: 

https://github.com/biomejs/biome/blob/47cb5d81f7f703b0ca149cda5e510dda6a0838f3/crates/biome_parser/src/lexer.rs#L84-L113